ql/src/java/org/apache/hadoop/hive/ql/lockmgr/HiveLockObject.java

```diff
@@ -85,7 +85,9 @@ public HiveLockObjectData(String data) {
       lockMode = elem[2];
       queryStr = StringInternUtils.internIfNotNull(elem[3]);
       if (elem.length >= 5) {
-        clientIp = elem[4];
```

Comment:
if we just do this above with split, will this work automatically?
```
      String[] elem = data.split(":", 5);
```

if this works we can avoid the prefix calculation & stuff like that

Reply:
Yes, this works. Good catch!